source/postprocess/dynamic_topography.cc

```diff
@@ -126,14 +138,30 @@ namespace aspect
                   const double dynamic_pressure   = in.pressure[q] - this->get_adiabatic_conditions().pressure(location);
                   const double sigma_rr           = gravity_direction * (shear_stress * gravity_direction) - dynamic_pressure;
                   const double dynamic_topography = - sigma_rr / gravity.norm() / density;
-
-                  integrated_topography += dynamic_topography * fe_values.JxW(q);
-                  integrated_surface_area += fe_values.JxW(q);
-
-                  stored_values.push_back (std::make_pair(location, dynamic_topography));
-                }
+          
+                 // JxW provides the volume quadrature weights. This is a general formulation
+                 // necessary for when a quadrature formula is used that has more than one point.
+                 dynamic_topography_x_volume += dynamic_topography * fe_values.JxW(q);
+                 volume += fe_values.JxW(q);  
+               }
+               
+               const double dynamic_topography_total = dynamic_topography_x_volume / volume;
```

Comment:
The name `dynamic_topography_x_volume` is perfectly chosen. It's the name `dynamic_topography_total` that I'm not happy with -- "total" is just not very descriptive. The variable computes a local average over the cell.